services/QuillLMS/spec/workers/teacher_requested_verification_email_worker_spec.rb

```diff
@@ -14,5 +14,19 @@
       expect_any_instance_of(UserMailer).to receive(:teacher_requested_verification_email).with(teacher, admin, message)
       subject.perform(admin.id, teacher.id, message)
     end
+
+    it 'should call refresh_token! on the admin and enqueue a token expiration job if they do not already have a token' do
+      expect_any_instance_of(User).to receive(:refresh_token!)
+      expect(ExpirePasswordTokenWorker).to receive(:perform_in).with(30.days, admin.id)
+
+      subject.perform(admin.id, teacher.id, message)
+    end
+
+    it 'should call refresh_token! on the admin if they do not already have a token' do
+      admin.refresh_token!
+
+      expect_any_instance_of(User).not_to receive(:refresh_token!)
+      subject.perform(admin.id, teacher.id, message)
+    end
```

Comment:
The linter may already be complaining about this, but `expect_any_instance_of` is generally frowned upon.  The "more approved" approach is something like this:
```ruby
before do
  allow(User).to receive(:find_by_id).and_return(admin, teacher)
end

it 'should call refresh_token! on the admin and enqueue a token expiration job if they do not already have a token' do
  expect(admin).to receive(:refresh_token!)
  expect(ExpirePasswordTokenWorker).to receive(:perform_in).with(30.days, admin.id)

  subject.perform(admin.id, teacher.id, message)
end
```
That `before` block should work for both `it` blocks.

Reply:
This is good to know, thanks! I didn't like the `expect_any_instance_of` here but was having a hard time figuring out a way around it.